File: frontend/src/api/system/flowNode.ts
import request from '@/utils/request'

/** 流程节点接口定义 */
export interface FlowNode {
  id?: number
  nodeKey: string            // BPMN taskDefinitionKey
  nodeName: string           // 节点中文名
  nodeType: string           // userTask / serviceTask
  assignee?: string          // 办理人表达式
  candidateGroups?: string   // 候选组
  targetStatus?: number      // 目标状态
  formSection?: string       // 表单区块
  isSystem?: number          // 是否系统节点 0/1
  processType?: string       // 所属流程类型
  delegateExpression?: string // serviceTask 委托表达式
  rejectToEnd?: number        // 驳回时是否直接结束流程 0-否 1-是
  description?: string       // 说明
  createTime?: string
  updateTime?: string
}

/** 获取所有节点列表（可按流程类型过滤） */
export function getFlowNodeList(processType?: string) {
  return request({ url: '/v1/flow-nodes', method: 'get', params: processType ? { processType } : {} })
}

/** 获取可用于编排的业务节点（可按流程类型过滤） */
export function getFlowNodeUserTasks(processType?: string) {
  return request({ url: '/v1/flow-nodes/user-tasks', method: 'get', params: processType ? { processType } : {} })
}

/** 获取可编排的节点（userTask + serviceTask，可按流程类型过滤） */
export function getFlowNodeOrchestratable(processType?: string) {
  return request({ url: '/v1/flow-nodes/orchestratable', method: 'get', params: processType ? { processType } : {} })
}

/** 获取节点详情 */
export function getFlowNodeDetail(id: number) {
  return request({ url: `/v1/flow-nodes/${id}`, method: 'get' })
}

/** 创建节点 */
export function createFlowNode(data: FlowNode) {
  return request({ url: '/v1/flow-nodes', method: 'post', data })
}

/** 更新节点 */
export function updateFlowNode(id: number, data: Partial<FlowNode>) {
  return request({ url: `/v1/flow-nodes/${id}`, method: 'put', data })
}

/** 删除节点 */
export function deleteFlowNode(id: number) {
  return request({ url: `/v1/flow-nodes/${id}`, method: 'delete' })
}

File: backend/src/main/java/com/declaration/entity/FlowNode.java
package com.declaration.entity;

import com.baomidou.mybatisplus.annotation.*;
import lombok.Data;

import java.time.LocalDateTime;

/**
 * 流程节点库实体（全局共享）
 *
 * @author Administrator
 * @since 2026-06-25
 */
@Data
@TableName("flow_node")
public class FlowNode {

    @TableId(value = "id", type = IdType.AUTO)
    private Long id;

    /** BPMN taskDefinitionKey（如 deptAudit, materialSubmit） */
    private String nodeKey;

    /** 节点中文名 */
    private String nodeName;

    /** 节点类型: userTask / serviceTask */
    private String nodeType;

    /** 办理人表达式（如 ${starterId}） */
    private String assignee;

    /** 候选组（如 MATERIAL_AUDITOR），多个逗号分隔 */
    private String candidateGroups;

    /** 到达此节点时 declaration_form.status 值 */
    private Integer targetStatus;

    /** 对应前端表单区块: basic/material/supplement/invoiceAmount/invoice */
    private String formSection;

    /** 是否系统内置节点 0-否 1-是（不可删除） */
    private Integer isSystem;

    /** 节点说明 */
    private String description;

    /** 所属流程类型: declaration/remittance/taxRefund */
    private String processType;

    /** serviceTask 的委托表达式（如 ${declarationServiceTask}） */
    private String delegateExpression;

    /** 驳回时是否直接结束流程 0-否(回退) 1-是(结束) */
    private Integer rejectToEnd;

    @TableField(fill = FieldFill.INSERT)
    private LocalDateTime createTime;

    @TableField(fill = FieldFill.INSERT_UPDATE)
    private LocalDateTime updateTime;
}
